### ISSUE TITLE
standby_cluster-side failover

### ISSUE THREAD

**Issue by mbanck** [zalando/patroni#2189]
I tried to look through all the (open or closed) issues with "standby_cluster" in it and couldn't find a discussion about this, so here goes:

1. It is well-known that libpq application-side failover is possible by putting all the hosts of a cluster in the connection string (like `host=pg1,pg2,pg3`), along with `target_session_attrs=read-write`. On failover or switchover the application will reconnect to the new primary.
2. In principle, I think it is possible to have this work with a stand-alone replication standby, i.e. put all the hosts in the `primary_conninfo`, along with `target_sesion_attrs`. The latter is not something one can do on the `pg_basebackup` command line (unless maybe one passes the full DSN and not `-h pg1,pg2,pg3`, not sure), but one can edit `postgresql.auto.conf` (for v12+) as written by `pg_basebackup`, add `target_session_attrs=read-write` and restart the standby. When there is a switchover/failover of the cluster, the stand-alone standby will automatically switch to following the new leader.
3. Turning to Patroni, when setting up a standby cluster, is possible to put more than one host in the `standby_cluster/host:` config, like this:
```
dcs:
  standby_cluster:
    host: pg1,pg2,pg3
```
4. This will put the hosts correctly into `primary_conninfo`, however, `primary_conninfo` is (for v12+) managed in `postgresql.conf` and it does not have `target_session_attrs=read-write`; maybe this could just be added as it shouldn't hurt (or are there use-cases for as standby-cluster not replicating from a leader?). One can edit it in of course, but then one has to make sure Patroni doesn't overwrite it all the time.
5. The problem with using `host: pg1,pg2,pg3` is that the pgpass logic just copies that host string into the designated pgpass file, and then PostgreSQL does not recognize the host. Again, one can edit this, but in my experience Patroni overwrites this as well.
6. An alternative would be something like
```
dcs:
  standby_cluster:
    host:
    - pg1
    - pg2
    - pg3
```
However, that fails because Patroni then passes something like `['pg1, 'pg2, 'pg3']` as host to `pg_basebackup` which is not a valid hostname: `pg_basebackup: error: could not translate host name "['pg1'" to address: Name or service not known`
7. So either the pgpass or the `pg_basebackup` logic would need to be changed along with adding `primary_session_attrs=read-write` to `primary_conninfo`.
8. Anything else I'm missing?

In conclusion, it would be nice if a standby_cluster would support this as well, but this is currently not the case; this issue is a feature request to track this. Unless I am missing something and this inherently not possible?

**Comment by mbanck** [comment]
I've opened two PRs (#2192 and #2193) in order to change the behaviour here, in my (somewhat limited so far) tests, this makes the standby cluster leader happily follow a main cluster switchover.

**Comment by CyberDem0n** [comment]
@mbanck I think this one could be closed?

**Comment by mbanck** [comment]
Yes, this should now be working, thanks for the reminder

**Comment by aminebt** [comment]
> I've opened two PRs (#2192 and #2193) in order to change the behaviour here, in my (somewhat limited so far) tests, this makes the standby cluster leader happily follow a main cluster switchover.

follow up question on this - just for information.
context:  in the setup I'm working on, the standby cluster is replicating not from the primary cluster's leader but rather from primary cluster's _replica_. replication is going fine but in case of failover pg_rewind fails with "connection to server at <new-primary-IP>, port <port> failed: **session is read-only**" which I suspect is due to target_session_attrs set to 'read, **write**' (as implemented in previous referenced commit). 

so I'm wondering whether there's a way to override target_session_attrs to 'any' or some other value through patroni or zalando-operator configuration ? otherwise is this use-case not supported (having standby cluster replicating from a replica)?
